controllers/tab/settings/archiving/form/ArchivingForm.inc.php

```diff
@@ -39,6 +69,46 @@ function __construct($wizardMode = false) {
 	function getLocaleFieldNames() {
 		return array('lockssLicense', 'clockssLicense');
 	}
+
+	/**
+	 * @see Form::execute()
+	 * @param $request PKPRequest
+	 */
+	function execute($request) {
+		parent::execute($request);
+
+		$versionDao = DAORegistry::getDAO('VersionDAO');
+		$product = $versionDao->getCurrentVersion("plugins.generic", "pln", true);
+		$categoryDir = PLUGINS_PREFIX . 'generic';
+		if (isset($product)) {
+			$file = $product->getProduct();
+
+			$plugin =& PluginRegistry::_instantiatePlugin('generic', $categoryDir, $file, $product->getProductClassname());
+			if ($plugin && is_object($plugin)) {
+				if (isset($this->_data['enablePln'])) {
+					if (!$plugin->getEnabled()) {
+						if ($plugin->getCanEnable()) {
+							$plugin->setEnabled(true);
+							$user = $request->getUser();
+							$notificationManager = new NotificationManager();
+							$notificationManager->createTrivialNotification($user->getId(), NOTIFICATION_TYPE_PLUGIN_ENABLED, array('pluginName' => $plugin->getDisplayName()));
+						}
+					}
+				} else {
+					if ($plugin->getEnabled()) {
+						if ($request->checkCSRF() && $plugin->getCanDisable()) {
```

Comment:
Since this is a `Form` class, I think the CSRF check can be done more cleanly with a call to `addCheck`.

Reply:
Fixed